optimizer.go

```diff
@@ -2,44 +2,122 @@ package main
 
 import (
 	"database/sql"
+	"encoding/json"
 	"flag"
 	"fmt"
 	"log"
+	"strings"
+
+	"github.com/creasty/defaults"
+)
+
+const (
+	pdokNamespace = "098c4e26-6e36-5693-bae9-df35db0bee49"
 )
 
 func main() {
 	log.Println("Starting...")
 	sourceGeopackage := flag.String("s", "empty", "source geopackage")
+	serviceType := flag.String("service-type", "ows", "service type to optimize geopackage for")
+	config := flag.String("config", "", "optional JSON config for additional optimizations")
 
 	flag.Parse()
-	optimizeGeopackage(*sourceGeopackage)
+
+	switch *serviceType {
+	case "ows":
+		optimizeOWSGeopackage(*sourceGeopackage)
+	case "oaf":
+		optimizeOAFGeopackage(*sourceGeopackage, *config)
+	default:
+		log.Fatalf("invalid value for service-type: '%s'", *serviceType)
+	}
 }
 
-func optimizeGeopackage(sourceGeopackage string) {
-	log.Printf("Performing for geopackage: '%s'...\n", sourceGeopackage)
+func optimizeOAFGeopackage(sourceGeopackage string, config string) {
+	log.Printf("Performing OAF optimizations for geopackage: '%s'...\n", sourceGeopackage)
 	db := openDb(sourceGeopackage)
 	defer db.Close()
 
 	tableNames := getTableNames(db)
 
-	for _, tablename := range tableNames {
-		createPuuid(tablename, db)
-		createFuuid(tablename, db)
+	if config != "" {
+		var oafConfig OafConfig
+		err := json.Unmarshal([]byte(config), &oafConfig)
+		if err != nil {
+			log.Fatalf("cannot unmarshal oaf config: %s", err)
+		}
+		err = defaults.Set(&oafConfig)
+		if err != nil {
+			log.Fatalf("failed to set default config: %s", err)
+		}
+		for _, tableName := range tableNames {
+			if _, ok := oafConfig.Layers[tableName]; !ok {
+				log.Printf("WARNING: no config found for gpkg table '%s'", tableName)
+				continue
+			}
+			layerCfg := oafConfig.Layers[tableName]
+
+			for _, stmt := range layerCfg.SQLStatements {
```

Comment:
Misschien even een comment erbij dat we expliciet eerst de sql uitvoeren zodat je in geval dat nodig is hier maximale vrijheid hebt.